src/chainparams.cpp

```diff
@@ -1067,3 +1069,34 @@ void SelectParams(const std::string& network)
 void ClearCheckpoints(CChainParams &params) {
     params.checkpointData = {};
 }
+
+Res UpdateCheckpointsFromFile(CChainParams &params, const std::string &fileName) {
+    std::ifstream file(fileName);
+    if (!file.good()) {
+        return Res::Err("File \"%s\" does not exist.", fileName);
```

Comment:
Let's make the error message a bit more contexual and adaptable to all errors: 
`Could not read checkpoint file %s. Ensure it's exists and has read permissions`